polkadot/runtime/parachains/src/builder.rs

```diff
@@ -663,14 +673,18 @@ impl<T: paras_inherent::Config> BenchBuilder<T> {
 		inclusion::PendingAvailability::<T>::remove_all(None);
 
 		// We don't allow a core to have both disputes and be marked fully available at this block.
-		let cores = self.max_cores();
+		let max_cores = self.max_cores();
 		let used_cores =
 			(self.dispute_sessions.len() + self.backed_and_concluding_cores.len()) as u32;
-		assert!(used_cores <= cores);
+		assert!(used_cores <= max_cores);
+		let fill_claimqueue = self.fill_claimqueue;
 
 		// NOTE: there is an n+2 session delay for these actions to take effect.
 		// We are currently in Session 0, so these changes will take effect in Session 2.
 		Self::setup_para_ids(used_cores);
+		configuration::ActiveConfig::<T>::mutate(|c| {
+			c.coretime_cores = used_cores;
+		});
```

Comment:
Why didn't we need this before? Until this PR the same benchmarks were working without this change. Are we sure that the cores will be updated correctly in production?

Reply:
The assignment provider was changed. Previously the number of cores was determined by the number of legacy parachains. Now the number of cores is determined by the coretime chain, hence we need to set it explicitly in testing. For production we already take care of setting that number to the right value in the migration [here](https://github.com/paritytech/polkadot-sdk/blob/b5b4a199cfd9311a1597ec42afcf559234788319/polkadot/runtime/parachains/src/coretime/migration.rs#L130).

Reply:
Oh sorry, there's a comment but I forgot about it:

```
/// Note that this configuration is managed by the coretime chain.
```